packages/components/src/mobile/color-settings/palette.screen.native.js

```diff
@@ -57,6 +55,10 @@ const PaletteScreen = () => {
 	const currentSegmentColors = ! isGradientSegment
 		? defaultSettings.colors
 		: defaultSettings.gradients;
+	const allAvailableColors = useMobileGlobalStylesColors();
+	const allAvailableGradients = currentSegmentColors
+		.map( ( item ) => item.gradients || [] )
+		.reduce( ( acc, val ) => acc.concat( val ), [] );
```

Comment:
Nitpick: we could simplify this by using `flatMap`.
```suggestion
	const allAvailableGradients = currentSegmentColors
		.flatMap( ( { gradients } ) => gradients )
		.filter( Boolean );
```

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/57605/commits/32466822ae1da7b17fad76139832ea8f8dfb3354